File: lib/ios/ThaliCore/ThaliCoreTests/ThaliSingleInstanceTests.m
//
//  TestAppTests.m
//  ThaliCoreTests
//
//  Copyright (C) Microsoft. All rights reserved.
//  Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
//

#import <XCTest/XCTest.h>
#import <ThaliCore/ThaliCore.h>

@interface ThaliSingleInstanceTests : XCTestCase
@end

@implementation ThaliSingleInstanceTests
{
  THEAppContext *_app;
}

- (void)setUp {
    [super setUp];
    _app = [[THEAppContext alloc] init];
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
    _app = nil;
}

- (void)testCanStartListening {
  XCTAssertTrue([_app startListeningForAdvertisements]);
}

- (void)testStartListeningTwiceIsAnError {
  XCTAssertTrue([_app startListeningForAdvertisements]);
  XCTAssertFalse([_app startListeningForAdvertisements]);
}

- (void)testStopListeningIsNotAnError {
  XCTAssertTrue([_app stopListeningForAdvertisements]);
}

- (void)testStartAdvertising {
  XCTAssertTrue([_app startUpdateAdvertisingAndListening:4242]);
}

- (void)testStopAdvertisingIsNotAnError {
  XCTAssertTrue([_app stopAdvertisingAndListening]);
}

- (void)testStartAdvertisingTwiceIsNotAnError {
  XCTAssertTrue([_app startUpdateAdvertisingAndListening:4242]);
  XCTAssertTrue([_app startUpdateAdvertisingAndListening:4242]);
}

- (void)testInviteContextLength {
  NSString *uuid = [NSString stringWithFormat:@"%@:%llu", [[[NSUUID alloc] init] UUIDString], ULLONG_MAX];

  NSString *contextString = [NSString stringWithFormat:@"%@+%@", uuid, uuid];
  NSData *contextData = [contextString dataUsingEncoding:NSUTF8StringEncoding];
  XCTAssertTrue([contextData length] == 115);
}


@end

File: lib/ios/ThaliCore/ThaliCoreTests/ThaliSingleInstanceTests.swift
//
//  ThaliSingleInstanceTests.swift
//  ThaliCore
//
//  Created by Ilya Laryionau on 7/29/16.
//  Copyright © 2016 Thali. All rights reserved.
//

import Foundation
import XCTest
@testable import ThaliCore

final class ThaliSingleInstanceTests: XCTestCase {
    private var app: THEAppContext?

    override func setUp() {
        super.setUp()

        app = THEAppContext()
    }

    override func tearDown() {
        super.tearDown()

        app = nil
    }

    func testCanStartListening() {
        XCTAssertTrue(app?.startListeningForAdvertisements())
    }

    func testStartListeningTwiceIsAnError() {
        XCTAssertTrue(app?.startListeningForAdvertisements())
        XCTAssertFalse(app?.startListeningForAdvertisements())
    }

    func testStopListeningIsNotAnError() {
        XCTAssertTrue(app?.stopListeningForAdvertisements())
    }

    func testStartAdvertising() {
        XCTAssertTrue(app?.startUpdateAdvertisingAndListening(4242))
    }

    func testStopAdvertisingIsNotAnError() {
        XCTAssertTrue(app?.stopAdvertisingAndListening])
    }

    func testStartAdvertisingTwiceIsNotAnError() {
    XCTAssertTrue(app?.startUpdateAdvertisingAndListening(4242))
    XCTAssertTrue(app?.startUpdateAdvertisingAndListening(4242))
    }

    func testInviteContextLength() {
        let uuid = "\(NSUUID().UUIDString):\(ULLONG_MAX)"
        let contextString = "\(uuid)+\(uuid)"
        let contextData = contextString.dataUsingEncoding(NSUTF8StringEncoding)

        XCTAssertTrue(contextData.length == 115)
    }
}
